src/Dofs/ConstraintHandler.jl

```diff
@@ -1019,8 +1019,9 @@ function _add!(ch::ConstraintHandler, pdbc::PeriodicDirichlet, interpolation::In
     # Dof map for mirror dof => image dof
     dof_map = Dict{dof_map_t,dof_map_t}()
 
-    mirror_dofs = zeros(Int, ndofs_per_cell(ch.dh))
-     image_dofs = zeros(Int, ndofs_per_cell(ch.dh))
+    n = ndofs_per_cell(ch.dh, first(face_map).mirror[1])
+    mirror_dofs = zeros(Int, n)
+    image_dofs  = zeros(Int, n)
```

Comment:
I think this is not correct. Think e.g. about one side of the domain has a quad and the other side of the domain has a triangle.

Reply:
The `collect_periodic_faces` only works when the mirror and image faces are the same celltype anyways, so I think this is fine.

Reply:
I think this is fine too, but perhaps it is clearer (showing that we assume the same interpolation) if using 
```suggestion
    n = getnbasefunctions(interpolation)
    mirror_dofs = zeros(Int, n)
    image_dofs  = zeros(Int, n)
```
?